Exception in thread Thread-3:
Traceback (most recent call last):
  File "/usr/lib64/python3.9/threading.py", line 980, in _bootstrap_inner
    self.run()
  File "/usr/lib64/python3.9/threading.py", line 917, in run
    self._target(*self._args, **self._kwargs)
  File "/home/nyepes2/cs425/g06/src/mp4/worker.py", line 387, in handle_client
    run_job(client)
  File "/home/nyepes2/cs425/g06/src/mp4/worker.py", line 312, in run_job
    received_stream = decode_key_val(data)
  File "/home/nyepes2/cs425/g06/src/mp4/worker.py", line 55, in decode_key_val
    return json.loads(line)
  File "/usr/lib64/python3.9/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/usr/lib64/python3.9/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib64/python3.9/json/decoder.py", line 353, in raw_decode
    obj, end = self.scan_once(s, idx)
json.decoder.JSONDecodeError: Unterminated string starting at: line 1 column 22 (char 21)
Exception in thread Thread-2:
Traceback (most recent call last):
  File "/usr/lib64/python3.9/threading.py", line 980, in _bootstrap_inner
    self.run()
  File "/usr/lib64/python3.9/threading.py", line 917, in run
    self._target(*self._args, **self._kwargs)
  File "/home/nyepes2/cs425/g06/src/mp4/worker.py", line 258, in handle_output
    pipe_vms(job)
  File "/home/nyepes2/cs425/g06/src/mp4/worker.py", line 188, in pipe_vms
    last_merge = randomized_sync_log(local_processed_log, log_name, processed_data, last_merge)
  File "/home/nyepes2/cs425/g06/src/mp4/worker.py", line 84, in randomized_sync_log
    send_int(sender_sock, int(val[1]))
  File "/home/nyepes2/cs425/g06/src/mp4/worker.py", line 35, in send_int
    sock.sendall(int_val.to_bytes(4, byteorder="little"))
BrokenPipeError: [Errno 32] Broken pipe
Traceback (most recent call last):
  File "/home/nyepes2/cs425/g06/tests/mp4_demo1.py", line 15, in <module>
Exception ignored in: <module 'threading' from '/usr/lib64/python3.9/threading.py'>
Traceback (most recent call last):
  File "/usr/lib64/python3.9/threading.py", line 1477, in _shutdown
Traceback (most recent call last):
  File "/home/nyepes2/cs425/g06/src/mp4/worker.py", line 436, in <module>
    rain_storm_framework(machine_id, filter_pattern)
  File "/home/nyepes2/cs425/g06/src/mp4/framework.py", line 12, in rain_storm_framework
    start_server(int(sys.argv[1]))
  File "/home/nyepes2/cs425/g06/src/mp4/worker.py", line 424, in start_server
    lock.acquire()
KeyboardInterrupt: 
    for line in sys.stdin:  # Read from stdin line by line
KeyboardInterrupt
    client_socket, ip_address = server.accept()
  File "/usr/lib64/python3.9/socket.py", line 293, in accept
    fd, addr = self._accept()
KeyboardInterrupt
